modules/app/claim/propose/form/SubmitForm.ts

```diff
@@ -12,76 +12,80 @@ import { TimeFormat } from "@/modules/app/claim/propose/TimeFormat";
       return ToastSender.Error("The provided markdown must at least have 100 characters.");
     }
-    if (state.markdown.length >= 2500) {
+    if (editor.markdown.length >= 2500) {
       return ToastSender.Error("The provided markdown must not be longer than 2500 characters.");
     }
   }
 
   {
-    if (!state.labels || state.labels === "") {
+    if (!editor.labels || editor.labels === "") {
       return ToastSender.Error("The proposed claim must have at least one category label.");
     }
-    if (SplitList(state.labels).length > 4) {
+    if (SplitList(editor.labels).length > 4) {
       return ToastSender.Error("The proposed claim must not have more than four category labels.");
     }
-    if (HasDuplicate(SplitList(state.labels))) {
+    if (HasDuplicate(SplitList(editor.labels))) {
       return ToastSender.Error("The proposed claim must not have duplicated category labels.");
     }
   }
 
   {
-    if (!state.expiry || state.expiry === "") {
+    if (!editor.expiry || editor.expiry === "") {
       return ToastSender.Error("The provided expiry must not be empty.");
     }
-    if (!moment(state.expiry, TimeFormat, true).isValid()) {
+    if (!moment(editor.expiry, TimeFormat, true).isValid()) {
       return ToastSender.Error(`The provided expiry must be in the format ${TimeFormat}.`);
     }
+    if (moment(editor.expiry, TimeFormat).isBefore(moment())) {
+      return ToastSender.Error("The provided expiry must not be in past.");
+    }
   }
 
   {
-    if (!state.stake || state.stake === "") {
+    if (!editor.stake || editor.stake === "") {
       return ToastSender.Error("You must stake a minimum amount of reputation with your claim.");
     }
-    if (!inpPrt(state.stake)) {
+    if (!inpPrt(editor.stake)) {
       return ToastSender.Error("The format for staked reputation must be [number token].");
     }
-    if (!inpNum(state.stake)) {
+    if (!inpNum(editor.stake)) {
       return ToastSender.Error("The amount of staked reputation must be positive.");
     }
-    if (!inpTok(state.stake)) {
+    if (!inpTok(editor.stake)) {
       return ToastSender.Error("The amount of staked reputation must be denominated in a whitelisted token.");
     }
   }
 
   const req: PostCreateRequest = {
-    expiry: moment(state.expiry, TimeFormat, true).unix().toString(),
+    expiry: moment(editor.expiry, TimeFormat, true).unix().toString(),
     kind: "claim",
-    labels: SplitList(state.labels).join(","),
+    labels: SplitList(editor.labels).join(","),
     lifecycle: "propose",
     option: "true",
     parent: "",
-    stake: state.stake.split(" ")[0],
-    text: state.markdown,
-    token: state.stake.split(" ")[1],
+    stake: editor.stake.split(" ")[0],
+    text: editor.markdown,
+    token: editor.stake.split(" ")[1],
   };
 
   try {
     const [res] = await PostCreate(AuthStore.getState().auth.token, [req]);
     ToastSender.Success("Hooray, thy claim proposed milady!");
+    editor.delete();
```

Comment:
Here we also do make sure that the editor state is reset once a claim was proposed successfully. That way you can theoretically propose multiple claims and start with a fresh editor every time.